### PR TITLE
 [RDK-43827] - [POC] Add L2 testing for System Service- Phase 1-B

### PR DESCRIPTION
**Pull request by sumshank**
[POC] Add L2 testing for System Service- Phase 1-B

### CODE REVIEW

File: L2Tests/tests/SystemService_L2test.cpp [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+    status = jsonrpc.Subscribe<JsonObject>(JSON_TIMEOUT,
+                                           _T("onTemperatureThresholdChanged"),
+                                           &AsyncHandlerMock::onTemperatureThresholdChanged,
+                                           &async_handler);
+
+    EXPECT_EQ(Core::ERROR_NONE, status);
+
+    /*Set temperature threshold */
+
+    EXPECT_CALL(iarmBusImplMock, IARM_Bus_Call)
```
Comment (anand-ky) [inline L2Tests/tests/SystemService_L2test.cpp:294]: Can we split the get and set into different tests.

**Review by anand-ky — CHANGES_REQUESTED** [review]

File: docs/api/L2TestsAPI.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
+<!-- Generated automatically, DO NOT EDIT! -->
+<a name="head.L2Tests_API"></a>
+# L2Tests API
```
[the remark this answers is not in the document]
Comment (sumshank) [reply docs/api/L2TestsAPI.md:3]: Moved under Tests/L2Tests folder

File: L2Tests/L2TestsPlugin.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
```
[the remark this answers is not in the document]
Comment (sumshank) [reply L2Tests/L2TestsPlugin.json]: Done

File: L2Tests/tests/SystemService_L2test.cpp [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+/********************************************************
+************Test case Details **************************
+** 1. Start Log upload
+** 2. Subscribe for powerstate change
+** 3. Subscribe for LoguploadUpdates
+** 4. Trigger system power state change from ON -> DEEP_SLEEP
+** 5. Verify UPLOAD_ABORTED event triggered because of power state
+** 6. Verify Systemstate event triggered and verify the response
+*******************************************************/
+TEST_F(Systemservice_L2test,SystemServiceUploadLogsandSystempowerstatechange)
```
[the remark this answers is not in the document]
Comment (sumshank) [reply L2Tests/tests/SystemService_L2test.cpp]: Done

File: L2Tests/tests/SystemService_L2test.cpp (hunk 1 above)
Comment (sumshank) [reply L2Tests/tests/SystemService_L2test.cpp]: This was part of L2 sequence.. we can remove get API here as it is not making much difference

File: docs/api/L2TestsAPI.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (anand-ky) [reply docs/api/L2TestsAPI.md:3]: I see L2TestsAPI.md file is still under docs/api. Pls move it under L2Tests/L2Tests folder.

File: L2Tests/tests/SystemService_L2test.cpp (hunk 1 above)
Comment (anand-ky) [reply L2Tests/tests/SystemService_L2test.cpp]: ok sounds good.

File: Tests/L2Tests/L2Tests/tests/SystemService_L2test.cpp [hunk 5]
```diff
@@ -0,0 +1,444 @@
+#include <gtest/gtest.h>
```
Comment (anand-ky) [inline Tests/L2Tests/L2Tests/tests/SystemService_L2test.cpp:1]: Rename SystemService_L2test.cpp to SystemService_L2Test.cpp to match the Pascal case class name.

File: Tests/L2Tests/L2Tests/tests/UsbAccess_l2tests.cpp [hunk 6]
```diff
@@ -0,0 +1,12 @@
+#include <gtest/gtest.h>
```
Comment (anand-ky) [inline Tests/L2Tests/L2Tests/tests/UsbAccess_l2tests.cpp:1]: Also rename UsbAccess_l2tests.cpp to UsbAccess_L2Tests.cpp

**Review by anand-ky — CHANGES_REQUESTED** [review]
Pls check L2 Tests failure. If you want you can remove L2-Tests.yml file for now so that it doesn't block CI and merge the other changes.

File: docs/api/L2TestsAPI.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sumshank) [reply docs/api/L2TestsAPI.md]: Done

File: Tests/L2Tests/L2Tests/tests/SystemService_L2test.cpp (hunk 5 above)
Comment (sumshank) [reply Tests/L2Tests/L2Tests/tests/SystemService_L2test.cpp]: Done

**Comment by sumshank** [comment]
> Pls check L2 Tests failure. If you want you can remove L2-Tests.yml file for now so that it doesn't block CI and merge the other changes.

tests passed now

**Pull request merged**